### PR TITLE
Improve formatting and language

### LINKED ISSUE #378: Fix find command description
image

Current description does not seem to mention that we search on substrings. And the Hans Bo example is wrong

### PR DESCRIPTION
**Pull request by chrisjohntan**
- Fixes #378 
- Add link icon to make internal link more obvious to the reader
- Improve language used in: 
  - Find command description and examples
  - Addtag command
  - Deltag command
  - Cattag command
- Improve formatting of warnings

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -173,30 +173,31 @@ Examples:
...
 * `find n/bernice t/is1108` returns `Bernice Yu`<br>
-  ![result for 'find n/bernice t/is1108'](images/findBerniceIS1108Result.png){ width="750px" }
+  ![result for 'find n/bernice t/is1108'](images/findBerniceIS1108Result.png){ width="675px" }
 
 * `find t/floor t/run` returns all contacts with any tag containing `floor` or `run` <br>
-  ![result for 'find t/floor t/run'](images/findFloorRunResult.png){ width="750px" }
+  ![result for 'find t/floor t/run'](images/findFloorRunResult.png){ width="675px" }
 
-<div style="page-break-after: always;"></div>
+* `find n/alex n/bernice t/floor t/neighbour` returns all contacts that have names containing either `alex` or `bernice` **with** tags containing either `floor` or `neighbour`
```
Comment (chrisjohntan) [inline docs/UserGuide.md:200]: Please review this section especially

Comment (CYX22222003) [reply docs/UserGuide.md:200]: I believe the description now is much clearer

File: docs/UserGuide.md [hunk 2]
```diff
@@ -263,20 +264,18 @@ Format: `cattag t/TAG [t/MORE_TAGS]…​ CATEGORY`
     **Networking**| `network`                | <span style="color:LimeGreen"> Green </span>          
     **Mentorship**| `mentor`                 | <span style="color:HotPink"> Pink </span>
 
-* Multiple tags can be categorized to the same category at a time.
+* Multiple tags can be categorized to the same category using a single command.
 * All tags specified must be valid existing tags.
 * Attempts to set a tag to its current category will cause the whole command to be rejected.
 * Only one category is allowed to be entered per command, i.e. `cattag t/tag1 acads t/tag2 general` is not allowed.
+* Note: Newly created tags (by [`add`:link:](#adding-a-person-add) or [`addtag`:link:](#adding-tags-to-a-specific-person-addtag)) will have category `General` and colour `Grey` by default.
```
Comment (CYX22222003) [inline docs/UserGuide.md:271]: Good to specify default behaviour.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -173,30 +173,31 @@ Examples:
...
 * Only the specified fields are searched.
-* The search is case-insensitive. e.g. `hans` will match `Hans`
-* The order of the keywords does not matter. e.g. `Hans Bo` will match `Bo Hans`
-* Persons matching at least one keyword will be returned (i.e. `OR` search).
-  e.g. `n/Hans n/Bo` will return `Hans Gruber`, `Bo Yang`
+* The search is case-insensitive. e.g. `hans` will match `Hans`.
+* The order of the keywords does not matter, e.g. `find n/Hans n/Bo` is equivalent to `find n/Bo n/Hans`.
+* Persons matching at least one keyword will be returned (i.e. `OR` search),
+  e.g. `find n/Hans n/Bo` will match persons with names `Hans Gruber` or `Bo Yang`.
+* Keywords do not need to fully match the field, a partial match is sufficient, e.g. `bern` can match names like `Bernice` or `Bernard`
```
Comment (CYX22222003) [inline docs/UserGuide.md:187]: Good to mention that we search by keywords

File: docs/UserGuide.md [hunk 4]
```diff
@@ -173,30 +173,31 @@ Examples:
...
 Format: 
 
 `find PREFIX/KEYWORD [PREFIX/MORE_KEYWORDS]…​`
 
 * Only the specified fields are searched.
-* The search is case-insensitive. e.g. `hans` will match `Hans`
-* The order of the keywords does not matter. e.g. `Hans Bo` will match `Bo Hans`
-* Persons matching at least one keyword will be returned (i.e. `OR` search).
-  e.g. `n/Hans n/Bo` will return `Hans Gruber`, `Bo Yang`
+* The search is case-insensitive. e.g. `hans` will match `Hans`.
```
Comment (CYX22222003) [inline docs/UserGuide.md:183]: Case-sensitivity is also an important detail.

**Review by CYX22222003 — APPROVED** [review]
LGTM.

**Pull request merged**